src/mame/drivers/missb2.cpp

```diff
@@ -211,15 +205,16 @@ static ADDRESS_MAP_START( slave_map, AS_PROGRAM, 8, missb2_state )
 ADDRESS_MAP_END
 
 // Looks like the original bublbobl code modified to support the OKI M6295.
-
-static ADDRESS_MAP_START( sound_map, AS_PROGRAM, 8, missb2_state )
+// due to some really wacky bugs in the way the oki6295 was hacked in place, writes will happen to
+// many addresses other than 9000: 9000-9001, 0000-0001, 3827-3828, 44a8-44a9
+static ADDRESS_MAP_START( missb2_sound_map, AS_PROGRAM, 8, missb2_state )
 	AM_RANGE(0x0000, 0x7fff) AM_ROM
 	AM_RANGE(0x8000, 0x8fff) AM_RAM
-	AM_RANGE(0xa000, 0xa001) AM_DEVREADWRITE("ymsnd", ym3526_device, read, write)
-	AM_RANGE(0xb000, 0xb000) AM_DEVREAD("soundlatch", generic_latch_8_device, read) AM_WRITENOP // message for main cpu
-	AM_RANGE(0xb001, 0xb001) AM_READNOP AM_WRITE(bublbobl_sh_nmi_enable_w)  // bit 0: message pending for main cpu, bit 1: message pending for sound cpu
-	AM_RANGE(0xb002, 0xb002) AM_WRITE(bublbobl_sh_nmi_disable_w)
 	AM_RANGE(0x9000, 0x9000) AM_READWRITE(missb2_oki_r, missb2_oki_w) //AM_MIRROR(0x0FFF)
+	AM_RANGE(0xa000, 0xa001) AM_DEVREADWRITE("ymsnd", ym3526_device, read, write) AM_MIRROR(0x0FFE)
```

Comment:
please be consistent with case in hex numbers - you've put uppercase hex in the same lines as lowercase hex.